financial-connections/src/main/java/com/stripe/android/financialconnections/FinancialConnectionsSheet.kt

```diff
@@ -42,8 +42,31 @@ class FinancialConnectionsSheet internal constructor(
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     @Parcelize
     data class ElementsSessionContext(
+        val initializationMode: InitializationMode,
+        val amount: Long?,
+        val currency: String?,
         val linkMode: LinkMode?,
-    ) : Parcelable
+    ) : Parcelable {
+
+        val paymentIntentId: String?
+            get() = (initializationMode as? InitializationMode.PaymentIntent)?.paymentIntentId
+
+        val setupIntentId: String?
+            get() = (initializationMode as? InitializationMode.SetupIntent)?.setupIntentId
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        sealed interface InitializationMode : Parcelable {
+
+            @Parcelize
+            data class PaymentIntent(val paymentIntentId: String) : InitializationMode
```

Comment:
All of these need RestrictTo as well.

Reply:
Oh, good catch. Missed that among all the `android/os/Parcelable$Creator` lines in the API file.